applications/system/js_app/modules/js_textbox.c

```diff
@@ -0,0 +1,226 @@
+
+    mjs_return(mjs, MJS_UNDEFINED);
+}
+
+static void js_textbox_is_open(struct mjs* mjs) {
+    JsTextboxInst* textbox = get_this_ctx(mjs);
+    if(!check_arg_count(mjs, 0)) return;
+
+    mjs_return(mjs, mjs_mk_boolean(mjs, !!textbox->thread));
+}
+
+static void textbox_deinit(void* context) {
+    JsTextboxInst* textbox = context;
+    furi_thread_join(textbox->thread);
+    furi_thread_free(textbox->thread);
+    textbox->thread = NULL;
+
+    view_dispatcher_remove_view(textbox->view_dispatcher, 0);
+    view_dispatcher_free(textbox->view_dispatcher);
+    textbox->view_dispatcher = NULL;
+    furi_record_close(RECORD_GUI);
+
+    text_box_reset(textbox->text_box);
+    furi_string_reset(textbox->text);
+}
+
+static void textbox_callback(void* context, uint32_t arg) {
+    UNUSED(arg);
+    textbox_deinit(context);
+}
+
+static bool textbox_exit(void* context) {
+    JsTextboxInst* textbox = context;
+    view_dispatcher_stop(textbox->view_dispatcher);
+    furi_timer_pending_callback(textbox_callback, textbox, 0);
+    return true;
+}
+
+static int32_t textbox_thread(void* context) {
+    ViewDispatcher* view_dispatcher = context;
+    view_dispatcher_run(view_dispatcher);
+    return 0;
+}
+
+static void js_textbox_show(struct mjs* mjs) {
+    JsTextboxInst* textbox = get_this_ctx(mjs);
+    if(!check_arg_count(mjs, 0)) return;
+
+    Gui* gui = furi_record_open(RECORD_GUI);
+    textbox->view_dispatcher = view_dispatcher_alloc();
+    view_dispatcher_enable_queue(textbox->view_dispatcher);
+    view_dispatcher_add_view(textbox->view_dispatcher, 0, text_box_get_view(textbox->text_box));
+    view_dispatcher_set_event_callback_context(textbox->view_dispatcher, textbox);
+    view_dispatcher_set_navigation_event_callback(textbox->view_dispatcher, textbox_exit);
+    view_dispatcher_attach_to_gui(textbox->view_dispatcher, gui, ViewDispatcherTypeFullscreen);
+    view_dispatcher_switch_to_view(textbox->view_dispatcher, 0);
+
+    textbox->thread =
+        furi_thread_alloc_ex("JsTextbox", 1024, textbox_thread, textbox->view_dispatcher);
+    furi_thread_start(textbox->thread);
+
+    mjs_return(mjs, MJS_UNDEFINED);
+}
+
+static void js_textbox_close(struct mjs* mjs) {
+    JsTextboxInst* textbox = get_this_ctx(mjs);
+    if(!check_arg_count(mjs, 0)) return;
+
+    if(textbox->thread) {
+        view_dispatcher_stop(textbox->view_dispatcher);
+        textbox_deinit(textbox);
+    }
+
+    mjs_return(mjs, MJS_UNDEFINED);
+}
+
+static void* js_textbox_create(struct mjs* mjs, mjs_val_t* object) {
+    JsTextboxInst* textbox = malloc(sizeof(JsTextboxInst));
+    mjs_val_t textbox_obj = mjs_mk_object(mjs);
+    mjs_set(mjs, textbox_obj, INST_PROP_NAME, ~0, mjs_mk_foreign(mjs, textbox));
+    mjs_set(mjs, textbox_obj, "setConfig", ~0, MJS_MK_FN(js_textbox_set_config));
+    mjs_set(mjs, textbox_obj, "addText", ~0, MJS_MK_FN(js_textbox_add_text));
+    mjs_set(mjs, textbox_obj, "emptyText", ~0, MJS_MK_FN(js_textbox_empty_text));
```

Comment:
`clearText()` maybe? or just `clear()` even

Reply:
Since it's `addText()`, `clearText()` sounds more intuitive to me, but both work well enough